src/lib.rs

```diff
@@ -1264,6 +1277,80 @@ where
     }
 }
 
+/// A streaming iterator that flattens nested streaming iterators.
+#[derive(Debug)]
+pub struct Flatten<I> {
+    iter: I,
+    first: bool,
+}
+
+impl<I, J> StreamingIterator for Flatten<I>
+where
+    I: StreamingIteratorMut<Item = J>,
+    for<'a> J: StreamingIterator + 'a,
```

Comment:
This creates an awkward lifetime error in `fn get` on anything before Rust 1.63, which I believe was due to the migrate borrowck mode, removed in rust-lang/rust#95565. However, it works on older versions if we use `I::Item` without any lifetime constraints, rather than a separate `J` parameter.